### ISSUE TITLE
Error in FastBoot: "Cannot read property 'require' of null"

### ISSUE THREAD

**Issue by marcoow** [glimmerjs/glimmer-vm#1141]
We're currently seeing an error in the FastBoot tests in ember-simple-auth that originates in @glimmer/runtime:

```
/var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/vendor.js:58488
    var nodeURL = _nodeModule.module.require('url');
                                     ^

TypeError: Cannot read property 'require' of null
    at /var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/@glimmer/runtime.js:288:1
    at internalRequire (/var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/loader.js:54:1)
    at internalRequire (/var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/loader.js:50:1)
    at internalRequire (/var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/loader.js:50:1)
    at internalRequire (/var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/loader.js:50:1)
    at require (/var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/loader.js:87:1)
    at /var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/vendor.js:76572:1
    at /var/folders/cx/h0pn1kjn4971gtp4_5mrm3940000gn/T/broccoli-70503kLAdbxQD1YdI/out-423-broccoli_merge_trees/assets/vendor.js:76573:2
    at Script.runInContext (vm.js:127:20)
    at VMSandbox.eval (/Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/vm-sandbox.js:14:16)
    at /Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/ember-app.js:190:15
    at Array.forEach (<anonymous>)
    at EmberApp.loadAppFiles (/Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/ember-app.js:187:21)
    at EmberApp.retrieveSandboxedApp (/Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/ember-app.js:235:10)
    at new EmberApp (/Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/ember-app.js:61:21)
    at FastBoot._buildEmberApp (/Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/index.js:114:17)
    at new FastBoot (/Users/marcoow/Code/ember-simple-auth/node_modules/fastboot/src/index.js:52:10)
    at /Users/marcoow/Code/ember-simple-auth/node_modules/ember-cli-fastboot/index.js:335:29
    at Layer.handle [as handle_request] (/Users/marcoow/Code/ember-simple-auth/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/Users/marcoow/Code/ember-simple-auth/node_modules/express/lib/router/index.js:317:13)
    at /Users/marcoow/Code/ember-simple-auth/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/marcoow/Code/ember-simple-auth/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/marcoow/Code/ember-simple-auth/node_modules/express/lib/router/index.js:275:10)
    at /Users/marcoow/Code/ember-simple-auth/node_modules/ember-cli/lib/tasks/server/middleware/broccoli-watcher/index.js:54:11
    at watcher.then.errorHandler.buildError (/Users/marcoow/Code/ember-simple-auth/node_modules/broccoli-middleware/lib/watcher-middleware.js:35:7)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

I suspect this has to do with the recent refactoring (#1138) since specifically this line is failing:

https://github.com/glimmerjs/glimmer-vm/commit/47142cc7d393c2993a49586cf67d8be377dcb5dc#diff-33ba693279a12bb3f292beae96c25b03R43

The error can be reproduced by running the ember-simple-auth test app for the latest ember-beta (which I suppose contains the above change):

```
yarn workspace test-app test:one ember-beta --- ember s
```

**Comment by raido** [comment]
I have hit same thing over here: https://github.com/adopted-ember-addons/ember-page-title/pull/168 

`beta` and `canary` builds fail with this error.

**Comment by rwjblue** [comment]
Thank you for reporting @marcoow!

**Comment by kiwiupover** [comment]
@rwjblue how are things going?

**Comment by rwjblue** [comment]
I've finally reproduced the error over in https://github.com/emberjs/ember.js/pull/19123. Working on a fix here in glimmer-vm now...

**Comment by kiwiupover** [comment]
@rwjblue good on ya mate!

**Comment by rwjblue** [comment]
Fixed by #1155, working on backporting and landing in affected Ember versions.
